File: Objective-C/Sorting/sorting algorithms/InsertionSort.m
//
//  InsertionSort.m
//  Sorting
//
//  Created by Paul Kraft on 14.04.16.
//  Copyright © 2016 pauljohanneskraft. All rights reserved.
//

#import "InsertionSort.h"

@implementation InsertionSort

+ (void) sort:(NSMutableArray *)array fromLeft:(long)start toRight:(long)end {
    for( long i = start; i <= end; i++ ) {
        long k = start;
        while(k < i && [array objectAtIndex:k] < [array objectAtIndex:i]) {
            
            k++;
        }
        [self move:array from:i downTo:k];
    }
}

@end

File: Swift/Sort/sorting algorithms/InsertionSort.swift
//
//  InsertionSort.swift
//  Sort
//
//  Created by Paul Kraft on 21.12.15.
//  Copyright © 2015 Paul Kraft. All rights reserved.
//

func insertionSort<T: Comparable>(unsorted: [T]) -> (String, [T]) {
    var sorted : [T] = []
    for i in 0..<unsorted.count {
        sorted.append(unsorted[i]);
    }
    for i in 0..<sorted.count {
        var k = 0
        while(k < i && sorted[k] < sorted[i]) { k++ }
        sorted.insert(sorted.removeAtIndex(i), atIndex: k)
    }
    return ("InsertionSort", sorted)
}
